src/rulesets/Base/indexing.jl

```diff
@@ -1,3 +1,53 @@
+#####
+##### getindex(::Tuple)
+#####
+
+function frule((_, ẋ), ::typeof(getindex), x::Tuple, i::Integer)
+    return x[i], ẋ[i]
+end
+
+function frule((_, ẋ), ::typeof(getindex), x::Tuple, i)
+    y = x[i]
+    return y, Tangent{typeof(y)}(ẋ[i]...)
+end
+
+_tuple_N(::Type{<:Tuple{Vararg{<:Any, N}}}) where {N} = Val(N)
+
+function rrule(::typeof(getindex), x::T, i::Integer) where {T<:Tuple}
+    function getindex_back_1(dy)
+        dx = ntuple(j -> j == i ? dy : NoTangent(), _tuple_N(T))
+        return (NoTangent(), Tangent{T}(dx...), NoTangent())
+    end
+    return x[i], getindex_back_1
+end
+
+function rrule(::typeof(getindex), x::T, i::Integer) where {T<:NTuple{<:Any,<:Number}}
```

Comment:
```suggestion
# Special case for tuples of only numbers
function rrule(::typeof(getindex), x::T, i::Integer) where {T<:NTuple{<:Any,<:Number}}
```

Do we really need this though?
using zero rather than `bstractZero` in `@scalar_rule` was to hack around  map widening vectors into Vector{Any}  due to not having small unions.
This is not applicable in this case. 

So i think `getindex_back_1` is still fine